src/components/results/detail/helpers/DocumentViewer.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+  <div>
+    <v-alert
+      border="left"
+      color="red lighten-2"
+      v-if="error"
+    >
+      <i>Error loading preview</i>
+    </v-alert>
+    <v-alert
+      border="left"
+      color="blue lighten-4"
+      v-else-if="!srcUrl"
+    >
+      <i>Loading preview</i>
+      <v-progress-linear
+        :indeterminate="progress===0"
+        :value="$data.progress"
+      />
+    </v-alert>
+    <iframe
+      v-else
+      :src="srcUrl"
+      width="100%"
+      height="700"
+      :type="file.mimetype"
+    />
+  </div>
+</template>
+
+<script>
+
+import mime from 'mime';
+import GoldenRetriever from '@/helpers/GoldenRetriever';
+// import { getFileExtension } from '@/helpers/fileHelper';
+import getResourceURL from '@/helpers/resourceURL';
+
+export default {
+  created() {
+    this.fetch();
+  },
+  data() {
+    return {
+      error: false,
+      progress: 0,
+      srcUrlFromBlob: '',
+    };
+  },
+  props: {
+    file: {
+      type: Object,
+      required: true,
+    },
+    active: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  computed: {
+    srcUrl() {
+      // console.log(this.$props.file.mimetype, mime.getExtension(this.$props.file.mimetype));
+      switch (mime.getExtension(this.$props.file.mimetype)) {
+        case 'epub':
+          return `https://readium.web.app/?epub=${getResourceURL(this.file.hash)}`;
+        // case 'rtf': // rtf does not work for some reason
+        case 'docx':
+        case 'xlsx':
+        case 'odf':
+        case 'odt':
+          // return `https://docs.google.com/gview?embedded=true&url=${
+          return `https://view.officeapps.live.com/op/embed.aspx?src=${
+            getResourceURL(this.file.hash)}`;
+        default:
+          return this.$data.srcUrlFromBlob;
+      }
+    },
+  },
+  watch: {
+    active(active) {
+      if (active) {
+        this.error = undefined;
+        this.fetch();
+      } else if (this.retriever && this.$data.progress < 100) {
+        this.$data.srcUrlFromBlob = '';
+        this.retriever.cancel()
+          .then(() => {
+            this.$data.progress = 0;
+            this.retriever.off();
+            delete this.retriever;
+            this.$data.progress = 0;
+          });
+      }
+    },
+  },
+  methods: {
+    fetch() {
+      if (this.$data.srcUrlFromBlob) {
+        return Promise.resolve(this.$data.srcUrlFromBlob);
+      }
+      if (!this.file.mimetype === 'application/pdf') return Promise.resolve();
```

Comment:
It might be good to explain here that we assume browsers to have a built-in PDF viewer here (if that is indeed the case).

Reply:
In addition, it does seem that the whole GolderRetriever blob is here just because of PDF support. Perhaps we could find an iframe-based JS PDF viewer, so that we simply have 1 line for full PDF support? This is a nice-to-have, not a requirement (as we already have it working ehehe).

Reply:
yeah, good point. The retriever is indeed for pdfs, because otherwise the loading of the iframe is ugly. 
The loading of the epubs and docs is now also ugly, (As in: no information about loading progress,) but the viewers do not work with blobs, or i could not figure it out.